plugins/sfxr/sfxr.cpp

```diff
@@ -133,10 +133,18 @@ void SfxrSynth::resetSample( bool restart )
 		env_vol=0.0f;
 		env_stage=0;
 		env_time=0;
+
 		env_length[0]=(int)(s->m_attModel.value()*s->m_attModel.value()*100000.0f);
 		env_length[1]=(int)(s->m_holdModel.value()*s->m_holdModel.value()*100000.0f);
 		env_length[2]=(int)(s->m_decModel.value()*s->m_decModel.value()*100000.0f);
 
+		// These will be used as divisors later, let's avoid division by 0
+		for(int i = 0; i < 3; ++i)
+		{
+			env_length[i] = (env_length[i] == 0) ? 1 : env_length[i];
```

Comment:
I wonder what happens when the compiler sees something like `env_length[i] = env_length[i]` which is the other branch of the conditional.